packages/worker/native/src/lib.rs

```diff
@@ -340,7 +438,7 @@ fn worker_break_loop(mut cx: FunctionContext) -> JsResult<JsUndefined> {
         callback: callback.root(&mut cx),
     };
     if let Err(err) = worker.sender.blocking_send(request) {
-        callback_with_error(&mut cx, callback, err)?;
+        callback_with_error(&mut cx, callback, |cx| UNEXPECTED_ERROR.from_error(cx, err))?;
```

Comment:
Maybe make a convenience function `callback_with_unexpected_error` to avoid repeating this closure a bunch.